code/game/machinery/doors/door.dm

```diff
@@ -64,7 +64,7 @@
 		return
 	if(ismob(AM))
 		var/mob/B = AM
-		if(B.stat)
+		if(B.stat && istype(B, /mob/living/simple_animal/drone))

```

Comment:
If you're going to do two checks do the one that does the most filtering of the result set first i.e there are more dead people hitting the door than drones, so we filter it to drones first, then worry about checking if they're dead.